src/main/java/org/sonarlint/intellij/analysis/Analysis.java

```diff
@@ -216,37 +208,48 @@ private Summary analyzePerModule(AnalysisScope scope, ProgressIndicator indicato
     indicator.setText("Running SonarLint Analysis for " + scope.getDescription());
 
     var analyzer = getService(project, SonarLintAnalyzer.class);
-    var progressMonitor = new TaskProgressMonitor(indicator, project, () -> cancelled);
     var results = new LinkedHashMap<Module, ModuleAnalysisResult>();
-    RawFindingHandler rawFindingHandler;
+    var listAnalysis = new HashSet<AnalysisState>();
     try (var findingStreamer = new FindingStreamer(callback)) {
-      rawFindingHandler = new RawFindingHandler(findingStreamer, cachedFindings);
-
       for (var entry : scope.getFilesByModule().entrySet()) {
         var module = entry.getKey();
-        rawFindingHandler.setCurrentModule(module);
-        results.put(module, analyzer.analyzeModule(module, entry.getValue(), rawFindingHandler, progressMonitor));
+        var analysisId = UUID.randomUUID();
+        var analysisState = new AnalysisState(analysisId, findingStreamer, cachedFindings, entry.getValue(), module);
+        listAnalysis.add(analysisState);
+        results.put(module, analyzer.analyzeModule(module, entry.getValue(), analysisState, indicator));
         checkCanceled(indicator);
       }
     }
-    return summarize(scope, rawFindingHandler, results);
+    return summarize(scope, listAnalysis, results);
   }
 
-  private Summary summarize(AnalysisScope scope, RawFindingHandler rawFindingHandler, Map<Module, ModuleAnalysisResult> resultsByModule) {
+  private Summary summarize(AnalysisScope scope, Set<AnalysisState> listAnalysis, Map<Module, ModuleAnalysisResult> resultsByModule) {
     var allFailedFiles = resultsByModule.values().stream().flatMap(r -> r.failedFiles().stream()).collect(toSet());
-    var analyzedFiles = scope.getAllFilesToAnalyze();
-    var issuesPerAnalyzedFile = getFindingsPerAnalyzedFile(rawFindingHandler.getIssuesPerFile(), analyzedFiles);
-    var securityHotspotsPerAnalyzedFile = getFindingsPerAnalyzedFile(rawFindingHandler.getSecurityHotspotsPerFile(), analyzedFiles);
+    var issuesPerAnalyzedFile = getLiveFindingsPerAnalyzedFile(listAnalysis);
+    var securityHotspotsPerAnalyzedFile = getSecurityHotspotsPerAnalyzedFile(listAnalysis);
     var findings = new LiveFindings(issuesPerAnalyzedFile, securityHotspotsPerAnalyzedFile);
-    return new Summary(project, scope.getFilesByModule(), allFailedFiles, rawFindingHandler.getRawIssueCount(), findings);
+    var rawIssueCount = listAnalysis.stream().mapToInt(AnalysisState::getRawIssueCount).sum();
+    return new Summary(project, scope.getFilesByModule(), allFailedFiles, rawIssueCount, findings);
```

Comment:
There is dead code in `Summary`: `getReportedRuleKeys()` and the `reportedRuleKeys` field

Reply:
The equivalent of `Summary.logFoundIssues` is now done by the backend, see [here](https://github.com/SonarSource/sonarlint-core/commit/8a4350cea76afab029917e57a973691c844fe646#diff-4035cca526399b3825a5a53782572063e86ec313d63de9127c6eaa79e505773eR573)